fix(insights): long hogql query overflow in tag

rmrt1n opened the pull request:
## Problem
Reported in [#16964](https://github.com/PostHog/posthog/issues/16964).

## Changes
**old:**
![image](https://github.com/PostHog/posthog/assets/51780559/bfdedd4b-c25f-44b9-b5d9-a7d476eb6e58)


**new:**
![image](https://github.com/PostHog/posthog/assets/51780559/97312e83-cf56-420a-a222-5c8a71193b5d)



<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
manually

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

rmrt1n committed:
fix(insights): long hogql query overflow in tag

rmrt1n commented:
Hi, new contributor here. I was contemplating whether the user should be able to view the whole query in the tag or not, as it is also shown in the insights title. What do you think?

mariusandra commented:
Hey @rmrt1n this is already a solid UX improvement. Thanks!

If you want to improve this event more, then indeed, adding a simple HTML `title` attribute (or a `Popover`) would be the next step. Mostly because now some information about the breakdown can be lost. Bonus points if you can also edit the query in the popover :).

mariusandra commented on frontend/src/lib/lemon-ui/LemonTag/LemonTag.scss:
```diff
@@ -9,6 +9,7 @@
     color: var(--default);
     line-height: 1rem;
     white-space: nowrap;
+    max-width: 100%;
```
I'm actually not sure if we want to change the behaviour or `LemonTag`-s everywhere. It might be safer to add the `w-max` CSS class to the breakdown tag explicitly.

rmrt1n committed:
refactor(insights): set max-width in breakdown tag instead and added title attribute

rmrt1n commented:
I moved the max-width property to the `BreakdownTag` component instead of `LemonTag`, and added the title attribute. I wasn't sure if I changing `PropertyKeyInfo` to render a `Popover` for HogQL would affect other components that are using it.
